src/compiler/checker.ts

```diff
@@ -18359,9 +18359,9 @@ namespace ts {
                     if (relation === comparableRelation && target.flags & TypeFlags.Primitive) {
                         const constraints = sameMap((source as IntersectionType).types, t => t.flags & TypeFlags.Primitive ? t : getBaseConstraintOfType(t) || unknownType);

```

Comment:
I think the correct fix is to change this line to

```suggestion
                        const constraints = sameMap((source as IntersectionType).types, getBaseConstraintOrType);
```

Reply:
Makes sense. Can confirm is passing the test. Just pushed this change.